kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/commands/application/ApplicationCommandRegistry.kt

```diff
@@ -14,52 +17,49 @@ import com.kotlindiscord.kord.extensions.commands.application.slash.SlashCommand
 import com.kotlindiscord.kord.extensions.commands.application.user.UserCommand
 import com.kotlindiscord.kord.extensions.commands.converters.SlashCommandConverter
 import com.kotlindiscord.kord.extensions.i18n.TranslationsProvider
+import dev.kord.common.annotation.KordExperimental
+import dev.kord.common.annotation.KordUnsafe
 import dev.kord.common.entity.ApplicationCommandType
 import dev.kord.common.entity.Snowflake
 import dev.kord.core.Kord
-import dev.kord.core.behavior.createApplicationCommands
 import dev.kord.core.behavior.createChatInputCommand
 import dev.kord.core.behavior.createMessageCommand
 import dev.kord.core.behavior.createUserCommand
+import dev.kord.core.entity.Guild
 import dev.kord.core.event.interaction.ChatInputCommandInteractionCreateEvent
 import dev.kord.core.event.interaction.MessageCommandInteractionCreateEvent
 import dev.kord.core.event.interaction.UserCommandInteractionCreateEvent
 import dev.kord.rest.builder.interaction.*
-import dev.kord.rest.json.JsonErrorCode
 import dev.kord.rest.request.KtorRequestException
-import kotlinx.coroutines.flow.toList
+import mu.KLogger
 import mu.KotlinLogging
 import org.koin.core.component.KoinComponent
 import org.koin.core.component.inject
 import java.util.*
 
-/** Registry for all Discord application commands. **/
-public open class ApplicationCommandRegistry : KoinComponent {
-    private val logger = KotlinLogging.logger { }
+/**
+ * Abstraction layer between the [ExtensibleBot] and it's event handling.
+ *
+ * An implementing class is supposed register all commands and handle their discord events and lifecycles.
```

Comment:
This isn't a fantastic doc comment, and it's not specific about what commands this handles - because, as I keep saying, KordEx has other types of commands.

Something along the following lines might be better:

```
Abstract class representing common behavior for application command registries.

Deals with the registration and syncing of, and dispatching to, all application commands.
Subtypes should build their functionality on top of this type.

@see DefaultApplicationCommandRegistry
```

Something like that.